src/main/java/eggventory/commands/edit/EditStockCommand.java

```diff
@@ -0,0 +1,53 @@
+package eggventory.commands.edit;
+
+import eggventory.StockList;
+import eggventory.Storage;
+import eggventory.commands.Command;
+import eggventory.enums.CommandType;
+import eggventory.enums.Property;
+import eggventory.items.Stock;
+import eggventory.ui.Cli;
+
+/**
+ * Command objects for editing stocks
+ * Requires the stockCode (as listed by the system) of the stock.
+ */
+public class EditStockCommand extends Command {
+
+    private String stockCode;
+    private Property property; //Stores the property you want to edit
+    private String newValue; //Stores the newValue you want
+
+    /**
+     * Initializes all the attributes of the details of the stock to be edited.
+     * @param type The type of command.
+     * @param stockCode The unique String that identifies a Stock.
+     * @param property The property of the Stock that is to be changed.
+     * @param newValue The newValue of the property.
+     */
+    public EditStockCommand(CommandType type, String stockCode, Property property, String newValue) {
+        super(type);
+        this.stockCode = stockCode;
+        this.property = property;
+        this.newValue = newValue;
+    }
+
+    /**
+     * Executes the actual editing of the stock's property.
+     * @param list StockList containing all the StockTypes.
+     * @param cli  Cli object to display output to.
+     * @param storage  Storage object to handle saving and loading of any data.
+     * @return String of the output, for JUnit testing.
+     */
+    @Override
+    public String execute(StockList list, Cli cli, Storage storage) {
+        String output;
+        Stock edited = list.setStock(stockCode, property, newValue);
```

Comment:
ok so list.setstock returns a stock called "beforeedits" (stocklist, line 134), but here the stock is named "edited" and it prints the attributes of the edited stock as the new version. only one of these can be correct (i think bc of the way java works it probably passes the edited version no matter what) so some of the variables should be renamed

Reply:
Rebecca, the context is different. In list.setStock, I am returning the beforeedits stock because I intend to print 

I have updated your stock from: <beforeedits> to <edited>